### ISSUE TITLE
Pop up the video instead of going to the video page. 

### ISSUE THREAD

**Issue by an00bhav** [Mester/demo-day-vikings#95]
We need to pop up the video rather than leave the page and go to the youtube/soundcloud page.

**Comment by bdauer** [comment]
Summing up what I'd said on Slack:

I think most users are looking for music, not videos, and the popup would interfere with casual browsing. If it opens in a separate tab then someone could e.g. control-click a bunch of links to open them in background tabs and peruse through, trying each out. If the music pops up, that gets in the way of a quick scan.

Thoughts?

**Comment by an00bhav** [comment]
If our focus is to present our users to learn more about music, then we should:
1) Remove the link to youtube/soundcloud. Because I think right now when user clicks on that link, they are directed to the youtube/soundcloud page. 
2) Add more information about the music, because right now we are not showing any information about the music in the results, apart from the genre and upvotes. 

Thoughts?

**Comment by bdauer** [comment]
I think you pinned down the issue really well. The question is what we're trying to present to our users. My understanding was that mainly we want to help people to find new music. If they're finding new music, the main thing they'll want is to be able to listen to it, maybe find related music. At least that's what I would want first and foremost. Only if I really liked a band would I look for more information.

At this point what we really need to do is clarify who our intended user is and what they would like from the app.

**Comment by an00bhav** [comment]
Yes, the focus now should be in understanding what our users should be, and what we want to present it to them. 

1) A search engine to search /r/listentothis
2) discover music
3) Maybe listen to music (Do we want them to leave timeless music webapp and go to youtube/sc?)
4) See what people are talking about on the music that they are interested in
5) anything else?

**Comment by bdauer** [comment]
There are a lot of potential features, but if we're talking about primary features I think 1-3 covers it. We already have 1 and 2 covered and 3 works but takes people away from the page.

**Comment by an00bhav** [comment]
So should we go for it? Embed the video in a small pop up, and autoplay the video. The users can click anywhere outside to close the video and continue browsing the search results.

**Comment by bdauer** [comment]
I think that would negatively impact browsing and if we were to embed, we should use iframe or something similar without the popup functionality. Let's say someone wants to hear a song and continue browsing while listening. That seems like a very likely use case.

**Comment by an00bhav** [comment]
I agree, then instead of pop up, we can use iframe, but let's not show the iframe on the results page. Let's collapse all the iframes by default, and only expand when the user wants to?

**Comment by bdauer** [comment]
Sounds like a great idea. It will keep things tidy instead of cluttering up with all of the iframes , presumably also reducing load time, and they can expand as desired.

**Comment by an00bhav** [comment]
Great, I'll start working on it, now the requirements are clear.
